blocks/main_loop.py

```diff
@@ -156,16 +159,17 @@ def _check_finish_training(self):
         # the iteration the corresponding log record can be found only in
         # the previous row.
         if (self.log.current_row.training_finish_requested or
-                self.log.current_row.keyboard_interrupt_received or
-                self.log.previous_row.keyboard_interrupt_received):
+                self.log.current_row.interrupt_received or
+                self.log.previous_row.interrupt_received):
             raise TrainingFinish
 
-    def _handle_keyboard_interrupt(self, signal_number, frame):
+    def _handle_interrupt(self, signal_number, frame):
         # After receiving a first keyboard interrupt signal,
         # ignore all following ones.
```

Comment:
That's not true any more and in fact I do not know what the correct policy is. I guess we should catch the first interrupt and try to handle it gracefully, and with the second and all the following we should just finish immediately. What do you think?